src/main/java/org/elasticsearch/script/ScriptService.java

```diff
@@ -194,8 +202,8 @@ public void setClient(Client client) {
     }
 
     public void close() {
-        for (ScriptEngineService engineService : scriptEngines.values()) {
-            engineService.close();
+        for (ScriptEngineService scriptEngine : scriptEngines) {
```

Comment:
can you use `IOUtils.close()` here please it will close all even if there are exceptions